IAGrim/UI/Filters/Damage.cs

```diff
@@ -22,79 +15,47 @@ public partial class Damage : UserControl {
                     dmgTypes.Add("Physical");
-                }
 
-                if (dmgPiercing.Checked) {
+                if (dmgPiercing.Checked)
                     dmgTypes.Add("Pierce");
-                }
 
-                if (dmgFire.Checked) {
+                if (dmgFire.Checked)
                     dmgTypes.Add("Fire");
-                }
 
-                if (dmgCold.Checked) {
+                if (dmgCold.Checked)
                     dmgTypes.Add("Cold");
-                }
 
-                if (dmgLightning.Checked) {
+                if (dmgLightning.Checked)
                     dmgTypes.Add("Lightning");
-                }
 
-                if (dmgAether.Checked) {
+                if (dmgAether.Checked)
                     dmgTypes.Add("Aether");
-                }
 
-                if (dmgVitality.Checked) {
+                if (dmgVitality.Checked)
                     dmgTypes.Add("Life");
-                }
 
-                if (dmgChaos.Checked) {
+                if (dmgChaos.Checked)
                     dmgTypes.Add("Chaos");
-                }
 
-                if (dmgPoison.Checked) {
+                if (dmgAcid.Checked)
                     dmgTypes.Add("Poison");
-                }
 
-                if (dmgElemental.Checked) {
+                if (dmgElemental.Checked || dmgTypes.Contains("Fire") || dmgTypes.Contains("Cold") || dmgTypes.Contains("Lightning"))
                     dmgTypes.Add("Elemental");
-                }
-
-                if (totalDamage.Checked) {
-                    filters.Add(new[] { "offensiveTotalDamageModifier" });
-                }
-
-
-                foreach (var damageType in dmgTypes) {
-                    var isElemental = damageType.Equals("Fire") || damageType.Equals("Cold") ||
-                                      damageType.Equals("Lightning");
-
-                    if (isElemental) {
-                        filters.Add(new[]
-                        {
-                            $"offensive{damageType}",
-                            $"offensive{damageType}Modifier",
-                            "offensiveElemental",
-                            "offensiveElementalModifier",
-                            $"offensiveSlow{damageType}",
-                            $"offensiveSlow{damageType}Modifier"
-                        });
-                    }
-                    else {
-                        filters.Add(new[]
-                        {
-                            $"offensive{damageType}",
-                            $"offensive{damageType}Modifier",
-                            $"offensiveSlow{damageType}",
-                            $"offensiveSlow{damageType}Modifier"
-                        });
-                    }
-                }
+
+                if (totalDamage.Checked)
+                    filters.Add(new[] {"offensiveTotalDamageModifier"});
+
+                foreach (var damageType in dmgTypes)
```

Comment:
I think this might be introducing a bug.. it will now add "offensiveFire" separately from offensiveElemental.
IIRC, that becomes an "AND" query not an "OR".

"WHERE STAT IN (elemental, fire)" vs "WHERE STAT IN (fire) AND STAT IN (elemental)"

Reply:
I already merged this into master but I've not pushed it out to anyone yet.
Confirmed that Fire, Cold and Lightning filters no longer works on the current tip of master.